src/main/java/net/obvj/performetrics/util/DurationFormat.java

```diff
@@ -0,0 +1,193 @@
+            {
+                return DurationFormat.FULL.format(duration, printLegend);
+            }
+            if (duration.getMinutes() > 0)
+            {
+                return String.format(MINUTES_FORMAT, duration.getMinutes(), duration.getSeconds(),
+                        duration.getNanoseconds()) + legend(printLegend, MINUTES_LEGEND);
+            }
+            return String.format(SECONDS_FORMAT, duration.getSeconds(), duration.getNanoseconds())
+                    + legend(printLegend, SECONDS_LEGEND);
+        }
+
+    },
+
+    /**
+     * Formats a time duration in any of the following formats: {@code H:M:s:ns},
+     * {@code M:S:ns}, or {@code S.ns}, suppressing trailing zeros from the nanosecond part.
+     * <p>
+     * Examples:
+     * <ul>
+     * <li>{@code 0.001 second(s)}</li>
+     * <li>{@code 3.2 second(s)}</li>
+     * <li>{@code 15:00.00589 minute(s)}</li>
+     * </ul>
+     */
+    SHORTER
+    {
+        @Override
+        public String format(final Duration duration, boolean printLegend)
+        {
+            String format = removeTrailingZeros(SHORT.format(duration, false));
+
+            if (!printLegend)
+            {
+                return format;
+            }
+            if (duration.getHours() > 0)
+            {
+                return format + legend(true, HOURS_LEGEND);
+            }
+            if (duration.getMinutes() > 0)
+            {
+                return format + legend(true, MINUTES_LEGEND);
+            }
+            return format + legend(true, SECONDS_LEGEND);
+        }
+
+    },
+
+    /**
+     * Formats a time duration using ISO-8601 seconds based representation, such as
+     * {@code PT8H6M12.345S}.
+     * <p>
+     * Examples:
+     * <ul>
+     * <li>{@code PT0.001S}</li>
+     * <li>{@code PT3.2S}</li>
+     * <li>{@code PT15M0.00589S}</li>
+     * </ul>
+     */
+    ISO_8601
+    {
+        @Override
+        public String format(final Duration duration, boolean printLegend)
+        {
+            if (Duration.ZERO.equals(duration))
+            {
+                return "PT0S";
+            }
+            StringBuilder builder = new StringBuilder();
+            builder.append("PT");
+            if (duration.getHours() > 0)
+            {
+                builder.append(duration.getHours()).append('H');
+            }
+            if (duration.getMinutes() > 0)
+            {
+                builder.append(duration.getMinutes()).append('M');
+            }
+            if (duration.getSeconds() > 0 || duration.getNanoseconds() > 0)
+            {
+                builder.append(duration.getSeconds());
+                if (duration.getNanoseconds() > 0)
+                {
+                    String nanos = removeTrailingZeros(String.format(NANOSECONDS_FORMAT, duration.getNanoseconds()));
+                    builder.append(".").append(nanos);
+                }
+                builder.append('S');
+            }
+            return builder.toString();
```

Comment:
Check if this is really needed now that we have the internal duration available.